### PR TITLE
fix: catch trailing whitespace inside inline HTML

### LINKED ISSUE #95: Research: PHPCS trailing-whitespace coverage for inline HTML
### Description
While investigating a recent commit, it was noted that `git diff --check` flagged trailing spaces in PHP files that should have been caught automatically by PHP_CodeSniffer (`phpcs`). 

Currently, developers have to manually run check commands or rely on Git to catch these spacing issues, which adds friction to the review workflow. We need to investigate why our `phpcs` configuration is failing to detect trailing whitespaces and fix it to automate this check.

### Expected Behavior
Running `phpcs` should flag any trailing whitespaces in PHP files and block the build/CI pipeline if they are present.

### Actual Behavior
`phpcs` passes successfully even when PHP files contain trailing spaces, forcing manual verification via `git diff --check`.

### Steps to Resolve
- Check the `phpcs.xml` configuration file to see which coding standard is being enforced.
- Verify if the `Generic.Files.LineEndings` or `Squiz.WhiteSpace.SuperfluousWhitespace` sniff is included and properly configured.
- Ensure that the trailing whitespace rules are not being explicitly excluded or overridden for PHP files.
- Update the ruleset to strictly forbid trailing whitespaces and verify the fix locally against a file known to have trailing spaces.

### Reference

- commit: https://github.com/fossasia/WPFAevent/pull/82/changes/c886104b28ac21f09f5090a857cd5d1c089065c8

### PR DESCRIPTION
**Pull request by arpan7sarkar**
## What

Adds a custom PHPCS sniff, `WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace`, that catches
trailing whitespace inside the inline-HTML regions of template files - the one place
`Squiz.WhiteSpace.SuperfluousWhitespace` cannot reach. Registers it in `phpcs.xml`, cleans
the 6 existing violations it found in `footer.php`, `header.php` and `page-past-events.php`
with `phpcbf`, and adds a regression test wired into the PHPCS CI job.

## Why

Closes #95. `Squiz.WhiteSpace.SuperfluousWhitespace`'s end-of-line check only listens for
`T_WHITESPACE`, a token the PHP tokenizer only emits inside `<?php ?>` blocks and doc
comments. Markup outside PHP tags - most of `public/partials/*.php` and
`public/templates/*.php` - is a single `T_INLINE_HTML` token per line, which that sniff never
registers, so trailing whitespace there passed `composer phpcs` with exit 0 even though
`git diff --check` flagged it. That's exactly what raised this, per the commit linked in the
issue.

I reproduced it directly against `main` before touching anything: a fixture with 3
trailing-whitespace lines inside `<?php ?>`-mixed HTML passed the existing ruleset clean
(0 errors, exit 0), confirming the gap wasn't a misconfiguration, just this sniff's fixed
token registration.

## How

Rather than trying to coerce configuration out of a sniff that fundamentally doesn't listen
for `T_INLINE_HTML`, this adds a small sniff scoped to exactly that token
(`phpcs/WPFAEvent/Sniffs/WhiteSpace/InlineHtmlTrailingWhitespaceSniff.php`). It only inspects
inline-HTML tokens that end in a newline, so it doesn't flag meaningful trailing space before
an embedded `<?php` tag, rtrims each one, and reports/auto-fixes through the standard
`addFixableError()` / `fixer` API PHPCS sniffs use. It deliberately leaves PHP-token
whitespace alone, so `Squiz.WhiteSpace.SuperfluousWhitespace` still owns that half and
there's no double-reporting.

## Testing

- `vendor/bin/phpcs` - exit 0 across the whole repo.
- `php tests/phpcs-whitespace-test.php` - the sniff's own regression test: builds a fixture
  with trailing whitespace after markup and a whitespace-only line inside markup, asserts
  both (and only those) are flagged, and asserts `phpcbf` cleans them without touching the
  rest of the file.
- `vendor/bin/phpstan analyse --memory-limit=2048M` - 2 pre-existing errors in
  `admin/class-wpfaevent-admin.php` (missing return types), confirmed present on `main` too
  and unrelated to this change.
- Manually reproduced the original bug against current `main` with a throwaway fixture
  (trailing whitespace inside `<?php ?>`-mixed HTML): passed clean before this change,
  correctly flagged on all 3 lines after.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Enforce consistent trailing-whitespace checks across both PHP code and inline HTML templates.

New Features:
- Add a PHPCS sniff that detects and auto-fixes trailing whitespace in inline HTML template markup.

Bug Fixes:
- Prevent trailing whitespace in inline HTML from bypassing the project coding standard.

Enhancements:
- Register the custom whitespace rule and document the separation from PHP-token whitespace checks.

CI:
- Run the inline-HTML whitespace regression test in the PHPCS CI workflow.

Tests:
- Add regression coverage for markup trailing whitespace, whitespace-only lines, and files without a final newline.

Chores:
- Clean existing inline-HTML trailing whitespace from event templates and add missing return annotations to affected admin methods.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved detection and automatic removal of trailing spaces and tabs in inline HTML, including files without a final newline.

* **Tests**
  * Added automated checks to verify whitespace violations are detected and corrected.
  * Integrated whitespace validation into the code quality workflow.

* **Style**
  * Cleaned up trailing and surrounding whitespace in header, footer, and past-event template markup without changing rendered content or behavior.

### CODE REVIEW

File: phpcs/WPFAEvent/Sniffs/WhiteSpace/InlineHtmlTrailingWhitespaceSniff.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		$eol     = $phpcsFile->eolChar;
+
+		// Without a newline the line continues into a PHP tag, where the
+		// whitespace before the tag is meaningful.
+		if ( substr( $content, - strlen( $eol ) ) !== $eol ) {
+			return;
+		}
+
+		$line    = substr( $content, 0, - strlen( $eol ) );
+		$trimmed = rtrim( $line, " \t" );
```
[the remark this answers is not in the document]
Comment (arpan7sarkar) [reply phpcs/WPFAEvent/Sniffs/WhiteSpace/InlineHtmlTrailingWhitespaceSniff.php]: Checked this against the actual tokenizer rather than the sniff logic in isolation: PHPCS splits a multi-line inline-HTML region into one `T_INLINE_HTML` token per line before any sniff runs, so `$content` in `process()` is never more than one line.

Reproduced your exact scenario directly:
```
$ printf '<?php\n$heading = "Hi";\n?>\n<div class="a">  \n\t\t\n</div>\n<?php\necho $heading;\n' > fixture.php
$ vendor/bin/phpcs --standard=phpcs.xml --sniffs=WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace fixture.php
 4 | ERROR | Whitespace found at end of line (WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace.Found)
 5 | ERROR | Whitespace found at end of line (WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace.Found)
```

Both lines 4 and 5 are flagged and auto-fixed independently, which is also what `tests/phpcs-whitespace-test.php` asserts and what a dedicated adversarial review pass confirmed by dumping the raw token stream. Not applying this suggestion - splitting `$content` on newlines would be redundant with what PHPCS already does, and would double-count lines that are already separate tokens.

(Separately: a real gap this same review pass did find - trailing whitespace on a file's last line when the file has no final newline at all - is fixed in e23626b.)

[the remark this answers is not in the document]
Comment (arpan7sarkar) [reply phpcs/WPFAEvent/Sniffs/WhiteSpace/InlineHtmlTrailingWhitespaceSniff.php:34]: Same claim CodeRabbit made on this PR, and it's still wrong for the same reason: PHPCS's tokenizer splits a multi-line inline-HTML region into one `T_INLINE_HTML` token per line before any sniff runs, so `$content` here is never more than one line.

Just re-verified against the current PR head (`93a23ab`, after the force-push) in a clean worktree:
```
$ printf '<?php\n$heading = "Hi";\n?>\n<div class="a">  \n\t\t\n</div>\n<?php\necho $heading;\n' > fixture.php
$ vendor/bin/phpcs --standard=phpcs.xml --sniffs=WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace fixture.php
 4 | ERROR | Whitespace found at end of line (WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace.Found)
 5 | ERROR | Whitespace found at end of line (WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace.Found)
```
Both lines 4 and 5 are flagged and auto-fixed independently - this is also exactly what `tests/phpcs-whitespace-test.php` asserts and runs in CI on every push. Not applying this suggestion; splitting `$content` on newlines would be redundant with what PHPCS already does.

File: phpcs/WPFAEvent/Sniffs/WhiteSpace/InlineHtmlTrailingWhitespaceSniff.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	public function register() {
+		return array( T_INLINE_HTML );
+	}
+
+	/**
+	 * Processes an inline HTML token.
+	 *
+	 * PHP_CodeSniffer emits one inline HTML token per line, and splits a line
+	 * into several tokens when PHP is embedded mid-line. Only the token that
+	 * carries the newline actually ends the line.
```
[the remark this answers is not in the document]
Comment (arpan7sarkar) [reply phpcs/WPFAEvent/Sniffs/WhiteSpace/InlineHtmlTrailingWhitespaceSniff.php]: The doc comment is accurate, not false - see the sibling thread on `process()` for the repro. PHPCS's tokenizer does emit one `T_INLINE_HTML` token per line (it splits a multi-line markup region during its own token-map pass, before any sniff runs), so the implementation isn't relying on incorrect behavior. Not changing this.

File: phpcs/WPFAEvent/Sniffs/WhiteSpace/InlineHtmlTrailingWhitespaceSniff.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+		if ( $trimmed === $line ) {
+			return;
+		}
+
+		$fix = $phpcsFile->addFixableError( 'Whitespace found at end of line', $stackPtr, 'Found' );
+
+		if ( true === $fix ) {
+			$phpcsFile->fixer->replaceToken( $stackPtr, $trimmed . ( $ends_with_eol ? $eol : '' ) );
+		}
```
[the remark this answers is not in the document]
Comment (arpan7sarkar) [reply phpcs/WPFAEvent/Sniffs/WhiteSpace/InlineHtmlTrailingWhitespaceSniff.php:44]: Same claim CodeRabbit made on this PR, and it's still wrong for the same reason: PHPCS's tokenizer splits a multi-line inline-HTML region into one `T_INLINE_HTML` token per line before any sniff runs, so `$content` here is never more than one line.

Just re-verified against the current PR head (`93a23ab`, after the force-push) in a clean worktree:
```
$ printf '<?php\n$heading = "Hi";\n?>\n<div class="a">  \n\t\t\n</div>\n<?php\necho $heading;\n' > fixture.php
$ vendor/bin/phpcs --standard=phpcs.xml --sniffs=WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace fixture.php
 4 | ERROR | Whitespace found at end of line (WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace.Found)
 5 | ERROR | Whitespace found at end of line (WPFAEvent.WhiteSpace.InlineHtmlTrailingWhitespace.Found)
```
Both lines 4 and 5 are flagged and auto-fixed independently - this is also exactly what `tests/phpcs-whitespace-test.php` asserts and runs in CI on every push. Not applying this suggestion; splitting `$content` on newlines would be redundant with what PHPCS already does.

**Review by Ubayed-Bin-Sufian — CHANGES_REQUESTED** [review]
I don't think this project needs a custom PHPCS sniff for this issue.

WPFAevent already has a repository-level whitespace check using `git diff --check`, so we already have a mechanism that enforces the actual repository requirement without depending on PHPCS, a specific IDE, or a specific editor.

Before adding a project-specific sniff, I think this issue should first be treated as a research task: investigate what PHPCS/WPCS already provides, what can be configured through the project `phpcs.xml` ruleset, and whether there is a limitation around inline HTML that cannot be handled through the existing rules.

If it cannot be solved with the existing PHPCS/WPCS functionality or configuration, please document that limitation in the issue.

The current PR adds a custom sniff, tests, and another CI check without establishing that this project actually needs any of them. That adds maintenance for something we already enforce at the repository level.

**I would suggest closing this PR and documenting the research/findings in the issue instead.**

**Comment by arpan7sarkar** [comment]
> I don't think this project needs a custom PHPCS sniff for this issue.
> 
> WPFAevent already has a repository-level whitespace check using `git diff --check`, so we already have a mechanism that enforces the actual repository requirement without depending on PHPCS, a specific IDE, or a specific editor.
> 
> Before adding a project-specific sniff, I think this issue should first be treated as a research task: investigate what PHPCS/WPCS already provides, what can be configured through the project `phpcs.xml` ruleset, and whether there is a limitation around inline HTML that cannot be handled through the existing rules.
> 
> If it cannot be solved with the existing PHPCS/WPCS functionality or configuration, please document that limitation in the issue.
> 
> The current PR adds a custom sniff, tests, and another CI check without establishing that this project actually needs any of them. That adds maintenance for something we already enforce at the repository level.
> 
> **I would suggest closing this PR and documenting the research/findings in the issue instead.**

Got it @Ubayed-Bin-Sufian  , I actully has done this one in 2 weeks ago when i was starting with the project then working with other better issues i forget to raise the pr that's why i raise this but yes now I have more context about the project so am closing the issue right now will invesitage what more improvment we can do to this project  and let you know
